### PR TITLE
Add testing section for individual/user testing 

### PR DESCRIPTION
**Pull request by s-christoff**
Howdy,
Some other Hashicorp products ([Waypoint](https://github.com/hashicorp/waypoint#running-the-unit-tests),[Consul](https://github.com/hashicorp/consul/blob/master/.github/CONTRIBUTING.md#testing),[Terraform](https://github.com/hashicorp/terraform/blob/master/.github/CONTRIBUTING.md#acceptance-tests-testing-interactions-with-external-services),[Nomad](https://github.com/hashicorp/nomad#testing),[Vault](https://github.com/hashicorp/vault#developing-vault), [Packer](https://github.com/hashicorp/packer/blob/master/.github/CONTRIBUTING.md#running-unit-tests),[Vagrant](https://github.com/hashicorp/vagrant/blob/main/.github/CONTRIBUTING.md#running-tests)) provide testing documentation in either their main readme or contributing documentation to help both operators and developers alike. We should do this too! 
I'm open to feedback on how to add a `make test` to the `Makefile` or something similar.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -148,6 +148,9 @@ website-start:
 test-ci: install-go
 	~/.go/bin/go test ./... -v $(TESTARGS) -timeout 120m
 
+test: 
+	~/.go/bin/go test ./... -timeout 120m
```
Comment (jimlambrt) [inline Makefile:152]: that's a REALLY long timeout :)

**Review by jimlambrt — APPROVED** [review]
Looks great!  Ty!

File: Makefile (hunk 1 above)
Comment (s-christoff) [reply Makefile:152]: I said the same thing - lol, seems on avg in CI our tests go about 30minutes. I can get it to 45minutes and see what happens?

Comment (jimlambrt) [reply Makefile:152]: I'm not sure I believe the CI numbers... it looks like they're done long before CI reports that they are done.   Can we try something like 30 min and see how it goes.   I really think, my average is more like 5 min when I run them without any caching.

**Pull request merged**